### ISSUE TITLE
wrong output type

### ISSUE THREAD

**Issue by danneggiato** [PowerShell/PowerShell#18519]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

the Get-GPInheritance commandlet returns wrong file type in GpoLinks and InheritedGpoLinks attributes. It shul return an array of Microsoft.GroupPolicy.GpoLink objects, but it returns an array of strings "Microsoft.GroupPolicy.GpoLink". In powershell 5.2 the same code produce the expected output

### Expected behavior

```console
PS> $(Get-GPInheritance -Target "ou=myou,dc=contoso,dc=com").InheritedGpoLinks

GpoId       : 
DisplayName : 
Enabled     : True
Enforced    : False
Target      : 
Order       : 1

GpoId       : 
DisplayName : 
Enabled     : True
Enforced    : False
Target      : 
Order       : 3
```

### Actual behavior

```console
PS> $(Get-GPInheritance -Target "ou=myou,dc=contoso,dc=com").InheritedGpoLinks
Microsoft.GroupPolicy.GpoLink
Microsoft.GroupPolicy.GpoLink
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jhoneill** [comment]
I think it is **returning** the objects of that type but the there is no formatting information to say how they should be printed so powerShell is **printing** the type name by default.  If you add  `| fl` or `| ft`  you should see all the properties.  You can add  type formatting so the output is always a table or list.

**Comment by danneggiato** [comment]
> I think it is **returning** the objects of that type but the there is no formatting information to say how they should be printed so powerShell is **printing** the type name by default. If you add `| fl` or `| ft` you should see all the properties. You can add type formatting so the output is always a table or list.

No, it returns an array of string that are all "Microsoft.GroupPolicy.GpoLink". If I add `| ft ` as you suggest nothing change. If I add `| gm` it says "TypeName: System.String". In Powershell 5 the same code returns the correct "TypeName: Microsoft.GroupPolicy.GpoLink".

**Comment by MichaelWinterGdB** [comment]
I can confirm this behavior for Powershell 7.3.1 and 7.3.2. as @danneggiato described.
Output on Powershell 5.1 gives the expected result.

**Comment by hao-public** [comment]
I can also confirm that this is an issue in PowerShell 7.

**Comment by marguz-wag** [comment]
Confirm this is in 7.3.4 as well.

![1686758994](https://github.com/PowerShell/PowerShell/assets/54038991/76212cb4-2ee6-4e23-9236-4a83eede116c)

**Comment by LainRobertson** [comment]
Noting the following article section:

- [about Windows PowerShell Compatibility - PowerShell | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_windows_powershell_compatibility?view=powershell-7.3#using-the-compatibility-feature)

It seems to be an issue with the serialisation and de-serialisation, which you can avoid through loading the module using the -SkipEditionCheck parameter:

`Import-Module -Name GroupPolicy -SkipEditionCheck`

I have not done any comprehensive testing as I don't use PowerShell (only Windows PowerShell), but as noted in more detail from the TechCommunity post that led here, -SkipEditionCheck brings the module in-process, thereby removing the relevance of serialisation/de-serialisation.

The bug should still be fixed but if the workaround provided by -SkipEditionCheck holds for the other commandlets, this might be "good enough" while you wait.

[Using 'Get-GPInheritance' Command in Powershell 7 - Microsoft Community Hub](https://techcommunity.microsoft.com/t5/windows-powershell/using-get-gpinheritance-command-in-powershell-7/m-p/2061936)

Cheers,
Lain

**Comment by korpx** [comment]
The issue still exist in 7.4.0

**Comment by SetiGeek** [comment]
Confirmed in 7.3.9
```
PS C:\> (Get-GPInheritance -target "...").gpolinks[0].getType()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     String                                   System.Object

PS C:\> $host.Version

Major  Minor  Build  Revision
-----  -----  -----  --------
7      3      9      -1
```

**Comment by mmseng** [comment]
Issue still exists in 7.5.3.
